Plugins/MatrixPlugin/MatrixForm.cs

```diff
@@ -0,0 +1,262 @@
+            List<string> DataSource = new List<string>(4) { (intBaseAttack).ToString(GlobalOptions.InvariantCultureInfo), (intBaseSleaze).ToString(GlobalOptions.InvariantCultureInfo), (intBaseDP).ToString(GlobalOptions.InvariantCultureInfo), (intBaseFirewall).ToString(GlobalOptions.InvariantCultureInfo) };
+
+            cboAttack.SuspendLayout();
+            cboSleaze.SuspendLayout();
+            cboDP.SuspendLayout();
+            cboFirewall.SuspendLayout();
+            cboAttack.BeginUpdate();
+            cboSleaze.BeginUpdate();
+            cboDP.BeginUpdate();
+            cboFirewall.BeginUpdate();
+
+            cboAttack.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboAttack.Enabled = false;
+            cboAttack.BindingContext = new BindingContext();
+            cboAttack.ValueMember = nameof(ListItem.Value);
+            cboAttack.DisplayMember = nameof(ListItem.Name);
+            cboAttack.DataSource = DataSource;
+            cboAttack.SelectedIndex = 0;
+            cboAttack.Visible = true;
+            cboAttack.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboAttack.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboSleaze.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboSleaze.Enabled = false;
+            cboSleaze.BindingContext = new BindingContext();
+            cboSleaze.ValueMember = nameof(ListItem.Value);
+            cboSleaze.DisplayMember = nameof(ListItem.Name);
+            cboSleaze.DataSource = DataSource;
+            cboSleaze.SelectedIndex = 1;
+            cboSleaze.Visible = true;
+            cboSleaze.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboSleaze.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboDP.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboDP.Enabled = false;
+            cboDP.BindingContext = new BindingContext();
+            cboDP.ValueMember = nameof(ListItem.Value);
+            cboDP.DisplayMember = nameof(ListItem.Name);
+            cboDP.DataSource = DataSource;
+            cboDP.SelectedIndex = 2;
+            cboDP.Visible = true;
+            cboDP.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboDP.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
+
+            cboFirewall.SelectedIndexChanged -= CbAttribute_SelectedIndexChanged;
+            cboFirewall.Enabled = false;
+            cboFirewall.BindingContext = new BindingContext();
+            cboFirewall.ValueMember = nameof(ListItem.Value);
+            cboFirewall.DisplayMember = nameof(ListItem.Name);
+            cboFirewall.DataSource = DataSource;
+            cboFirewall.SelectedIndex = 3;
+            cboFirewall.Visible = true;
+            cboFirewall.Enabled = objThis.CurrentPerson.CanSwapAttributes;
+            cboFirewall.SelectedIndexChanged += CbAttribute_SelectedIndexChanged;
```

Comment:
I use "ProcessMatrixAttributeComboBoxChange" function as example, one change i made is "ProcessMatrixAttributeComboBoxChange" shows matrix values with all modifiers (e.g. total)
my function shows only base attributes.
"ProcessMatrixAttributeComboBoxChange" doesn't contain "PopulateWithListItems" so i don't use this function too. And i don't where should i insert it.

Reply:
Yeah I gave you the wrong name, it is actually `WinFormsExtensions.PopulateWithListItem()`

You can find it in `WinFormsExtension.cs`
It sets the `ValueMember`, `DisplayMember` and `DataSource` properties of your cbo.

You would call it like this.

``` c#

List<ListItem> MyList = new List<ListItem>()

//Add stuff to this list here

cbo.Sleaze.PopulateWithListItem(MyList)
``` 

That replaced all of this.

``` c#
cboSleaze.BindingContext = new BindingContext();
cboSleaze.ValueMember = nameof(ListItem.Value);
cboSleaze.DisplayMember = nameof(ListItem.Name);
cboSleaze.DataSource = DataSource;
```
In addition setting a `DataSource` is very slow in WinForms and `PopulateWithListItem()` only sets it if it's really needed.

You can find an example in  `frmSelectLifestyle_Load()` in `frmSelectLifestyle.cs` line 54 onwards.

```c#
        public static void PopulateWithListItems(this ComboBox cboThis, IEnumerable<ListItem> lstItems)
        {
            if (ReferenceEquals(cboThis.DataSource, lstItems))
                return;
            // Binding multiple ComboBoxes to the same DataSource will also cause selected values to sync up between them.
            // Resetting bindings to prevent this though will also reset bindings to other properties, so that's not really an option
            // This means the code we use has to set the DataSources to new lists instead of the same one.
            List<ListItem> lstItemsToSet = lstItems?.ToList();
            if (!(cboThis.DataSource is IEnumerable<ListItem> lstCurrentList))
            {
                cboThis.ValueMember = nameof(ListItem.Value);
                cboThis.DisplayMember = nameof(ListItem.Name);
            }
            // Setting DataSource is slow because WinForms is old, so let's make sure we definitely need to do it
            else if (lstItemsToSet != null && lstCurrentList.SequenceEqual(lstItemsToSet))
                return;
            if (cboThis.DataSource != null)
                cboThis.BindingContext = new BindingContext();
            cboThis.DataSource = lstItemsToSet;
        }
```

Reply:
Thank you for detailed answer.
Add changes in https://github.com/chummer5a/chummer5a/pull/4563/commits/a243548b29d06e2009e2668e271c23b8f9ae635a